### DISCUSSION TITLE
How to know Webhook Fields

### DISCUSSION THREAD

**Discussion by NickCc6** [emqx/emqx#13313] [Q&A]
My first time using EMQX, I used Webhook to implement the business, but I couldn't find any documentation about the Webhook fields. Where should I get these field descriptions

If I add a webhook about connections, but I don't know what kind of data EMQX will send me

eg:
```{
"username": "undefined",
"topic": "glvac/GL764",
"timestamp": 1718961757271,
"qos": 0,
"publish_received_at": 1718961757270,
"pub_props": {
"User-Property": {}
},
"peerhost": "172.17.0.1",
"payload": "{\r\n \"msg\": \"hello GL764\"\r\n}",
"node": "emqx@172.17.0.3",
"metadata": {
"rule_id": "publish_msg_WH_D"
},
"id": "00061B62F8ECB2224259000021CE0012",
"flags": {
"retain": false,
"dup": false
},
"event": "message.publish",
"clientid": "MzE3MDkxMDQ2MDc0NjA0OTMzNTM1MjQxMjg5ODU3NzYxMjI"
}```

**Comment by zmstone** [comment]
Hi, there is no formal document for the data fields because it's highly composable.
as you can see in this doc, the default compose is to use `${payload}`, which in your case, it's `"{\r\n \"msg\": \"hello GL764\"\r\n}"`.
https://docs.emqx.com/en/enterprise/latest/dashboard/bridge.html#webhook

**Reply by NickCc6** [reply]
Thanks for your help

**Comment by keynslug** [comment]
I guess you were looking for https://docs.emqx.com/en/enterprise/latest/data-integration/rule-sql-events-and-fields.html#connection-complete-event-events-client-connected, which has of them described, although a bit briefly.

**Reply by NickCc6** [reply]
yes, yes, that's exactly what I want

thank you！
